tests/crosslink.e2e.ts

```diff
@@ -0,0 +1,92 @@
+import { expect, Page, test } from '@playwright/test';
+import { gotoHtmlWidget } from './utlis/server/playground';
+import { prepareNetwork, printlnCode } from './utlis';
+import { OPEN_EDITOR_SELECTOR, WIDGET_SELECTOR } from './utlis/selectors';
+
+test.describe('open in playground', () => {
+  test.beforeEach(async ({ page, baseURL }) => {
+    await prepareNetwork(page, baseURL); // offline mode
+  });
+
+  test('default', async ({ page }) => {
+    test.slow();
+
+    const code = printlnCode('Hello, world!');
+    await gotoHtmlWidget(page, { selector: 'code' }, `<code>${code}</code>`);
+    const editor = page.locator(WIDGET_SELECTOR);
+
+    const link = editor.locator(OPEN_EDITOR_SELECTOR).locator('a[href]');
+
+    const url =
+      'https://play.kotlinlang.org/editor/v1/N4Igxg9gJgpiBcIBmBXAdgAgLYEMCWaAFAJQbAA6mG1ADgE4EAuANkeSABIzPMQA0GAO4Q6zKAEJ2xStQC%2BIPiEY46AcxiMACsxyMkIrAhAArHADccC8BCw08zGHQBqjgM54IaIwEYAdAA5fACZvEFkgA%3D%3D%3D';
+
+    await Promise.all([
+      expect(link).toHaveCount(1),
+      expect(link).toHaveAttribute('target', '_blank'),
+      expect(link).toHaveAttribute('rel', 'noopener noreferrer'),
+      expect(link).toHaveAttribute('href', url),
+    ]);
+
+    const [newPage] = await Promise.all([
+      page.context().waitForEvent('page'), // open new page with target blank
+      link.first().click(),
+    ]);
+
+    expect(newPage.url()).toEqual(url); // new page with correct url
+
+    // wait until final page with editor
+    await newPage.waitForNavigation({
+      url: (url) => url.toString().startsWith('https://play.kotlinlang.org/#'),
+    });
+
+    // check it exists on page
+    await expect(newPage.locator('body')).toContainText(code);
+
+    await newPage.close();
+  });
+
+  test('hidden for crosslink=disabled', ({ page }) =>
+    checkLinkIsHidden(
+      page,
+      `<code data-crosslink="disabled">Hello, World!</code>`,
+    ));
+
+  test('no link for js-libs', ({ page }) => {
+    test.fixme(
+      true,
+      "Test doesn't work, BUG in code! jsLibs.length -> jsLibs.size",
```

Comment:
Lol! I found a real bug! 🫣

There is [**no** `.length`](https://github.com/JetBrains/kotlin-playground/blob/d4d4d1bc6f5ffd39d980430413b11a50fa1f512b/src/executable-code/index.js#L127) for [Set](https://github.com/JetBrains/kotlin-playground/blob/d4d4d1bc6f5ffd39d980430413b11a50fa1f512b/src/executable-code/index.js#L217)!